pkg/cortex/cortex.go

```diff
@@ -170,6 +172,7 @@ func (c *Config) RegisterFlags(f *flag.FlagSet) {
 	c.MemberlistKV.RegisterFlags(f)
 	c.QueryScheduler.RegisterFlags(f)
 	c.Tracing.RegisterFlags(f)
+	c.ResourceThresholds.RegisterFlags(f)
```

Comment:
Just a Nit: I don't like the naming. Actually I prefer `resource_monitor` as a name as resource threshold seems just used for throttling, which is a bit limited.